### PR TITLE
GHC 9.2.4 preparations

### PR DESCRIPTION
**Pull request by newhoggy**
These changes move the code towards `ghc-9.2.4` compatibility.

### CODE REVIEW

File: cardano-api/src/Cardano/Api/Query.hs [hunk 1]
```diff
@@ -381,7 +381,6 @@ newtype CurrentEpochState era = CurrentEpochState (Shelley.EpochState (ShelleyLe
 
 decodeCurrentEpochState
   :: forall era. Ledger.Era (ShelleyLedgerEra era)
-  => Share (Core.TxOut (ShelleyLedgerEra era)) ~ Interns (Shelley.Credential 'Shelley.Staking (Ledger.Crypto (ShelleyLedgerEra era)))
```
Comment (Jimbo4350) [inline cardano-api/src/Cardano/Api/Query.hs:384]: Interesting, what prompted this change?

File: cardano-tracer/cardano-tracer.cabal [hunk 2]
```diff
@@ -123,7 +123,6 @@ library
                        , async-extras
                        , bimap
                        , blaze-html
-                       , blaze-markup
```
Comment (Jimbo4350) [inline cardano-tracer/cardano-tracer.cabal:126]: :+1:

**Review by Jimbo4350 — APPROVED** [review]

File: cardano-api/src/Cardano/Api/Query.hs (hunk 1 above)
Comment (newhoggy) [reply cardano-api/src/Cardano/Api/Query.hs:384]: `ghc-9.2.4` flagged that as an unused constrained and given that it compiles with `ghc-8.10.7` flagging it as unused is correct.

**Comment by newhoggy** [comment]
I've verified the Windows CI build failures are unrelated to this PR:

* https://github.com/input-output-hk/cardano-node/pull/4505
* https://github.com/input-output-hk/cardano-node/actions/runs/3185450606/jobs/5195035946

**Review by erikd — APPROVED** [review]

**Review by MarcFontaine — APPROVED** [review]
LGTM Thanks

**Review by mgmeier — APPROVED** [review]

**Comment by newhoggy** [comment]
Squashed commits as requested.

**Review by deepfire — APPROVED** [review]

**Review by denisshevchenko — APPROVED** [review]

**Pull request merged**